crates/turborepo-vt100/src/entire_screen.rs

```diff
@@ -44,6 +27,53 @@ impl<'a> EntireScreen<'a> {
         s
     }
 
+    /// Returns the formatted contents of the terminal by row,
+    /// restricted to the given subset of columns.
+    ///
+    /// Formatting information will be included inline as terminal escape
+    /// codes. The result will be suitable for feeding directly to a raw
+    /// terminal parser, and will result in the same visual output.
+    ///
+    /// You are responsible for positioning the cursor before printing each
+    /// row, and the final cursor position after displaying each row is
+    /// unspecified.
+    // the unwraps in this method shouldn't be reachable
+    #[allow(clippy::missing_panics_doc)]
+    pub fn rows_formatted(
+        &self,
+        start: u16,
+        width: u16,
+    ) -> impl Iterator<Item = Vec<u8>> + '_ {
+        let mut wrapping = false;
+        let grid = self.screen.grid();
+        let (rows, _) = self.size();
+        let default = crate::attrs::Attrs::default();
+        grid.all_rows().take(rows).enumerate().map(move |(i, row)| {
+            // number of rows in a grid is stored in a u16 (see Size), so
+            // visible_rows can never return enough rows to overflow here
+            let i = i.try_into().unwrap();
+            let mut contents = vec![];
+            // We don't need final cursor position as long as CRLF is used and not just LF
```

Comment:
I don't understand this comment. Where would the CRLF or LF be coming from? is it user input? (in which case, is LF not possible on Windows?)

Reply:
When a terminal is in raw mode a LF only will move the cursor down a row, but not return it to the home (zero) column. This leads the string `foo\nbar` being printed in a raw terminal looking like
```
foo
   bar
```
where `foo\r\nbar` will be printed as expected
```
foo
bar
```

The comment is meant to explain why we aren't storing/restoring the final cursor position as we expect the caller to print these lines with a trailing `\r\n`.